modules/apps/accessibility/accessibility-menu-web/src/main/resources/META-INF/resources/js/AccessibilityMenu.js

```diff
@@ -30,28 +32,10 @@ import {getSettingValue, toggleClassName} from './util';
 
 const OPEN_ACCESSIBILITY_MENU_EVENT_NAME = 'openAccessibilityMenu';
 
-const getInitialSettings = (settings) => {
-	return settings.map((setting) => {
-		const {
-			className,
-			defaultValue,
-			key,
-			label,
-			sessionClicksValue,
-		} = setting;
-
-		const value = getSettingValue(defaultValue, sessionClicksValue, key);
-
-		toggleClassName(className, value);
-
-		return {className, key, label, value};
-	});
-};
+export const accessibilityMenuAtom = State.atom('accessibility-menu', []);
 
 const AccessibilityMenu = (props) => {
-	const [settings, setSettings] = useState(() =>
-		getInitialSettings(props.settings)
-	);
+	const [settings, setSettings] = useLiferayState(accessibilityMenuAtom);
```

Comment:
Well, maybe so, but I think it's a bit exaggerated at this point if you're going to think about performance, it's a very small list and using both approaches we won't have any perceptible improvement in that regard. I think it would be more a matter of convenience if that were the case in adopting this.

In that case the `settingKeys` could be a `useMemo` on the first render of the component or just calling it directly on render would also have no negative performance effects.

Reply:
Yeah, I wasn't thinking about it as a performance improvement. I was thinking more that since we are adding this to LiferayState, it ends up being our api for accessing settings. It's a much simpler api to have direct access to the setting based on it's key rather than having to loop through the settings to find the setting you're looking for.

Reply:
Got it, I think if we're just thinking about simplifying the public API it makes sense, performance we won't see any optimization effect even if it's looping to find or update.